lib/plugins/aws/package/compile/events/sqs/index.js

```diff
@@ -12,9 +12,20 @@ class AwsCompileSQSEvents {
       'package:compileEvents': this.compileSQSEvents.bind(this),
     };
 
-    // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8033
     this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'sqs', {
-      anyOf: [{ type: 'string' }, { type: 'object' }],
+      oneOf: [
+        { $ref: '#/definitions/awsArn' },
+        {
+          type: 'object',
+          properties: {
+            arn: { $ref: '#/definitions/awsArn' },
+            batchSize: { type: 'integer' },
```

Comment:
We should configure `minimum` and `maximum` constraints as indicated in AWS doc: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-eventsourcemapping.html#cfn-lambda-eventsourcemapping-batchsize

Reply:
Thanks @medikoo .
Regarding removing inline validation - should I also remove all the tests that test those validations?
I assumed they would still apply (expecting ajv to now throw the errors instead of the inline validations), but it seems to me the tests aren't invoking the ajv validation. Meaning, I messed up the schema deliberately and expected that the tests will fail with ajv errors but they didn't.

Reply:
> Regarding removing inline validation - should I also remove all the tests that test those validations?

Yes, definitely :)

Reply:
> but it seems to me the tests aren't invoking the ajv validation

Yes, it's a problem with tests. Ideally they should be written with help of [runServerless](https://github.com/serverless/serverless/tree/master/test#unit-tests) util, but it's old tests, that were introduced before we've changed the approach.

We would welcome the PR that refactors those tests, still  that should be not  in scope of this PR.

Also we agreed to not test that schema validation works (it increases maintenance cost and seems a bit superfluous when schema validation engine is extensively tested on its own). In that context we should test just valid scenarios, and that already makes a great tests that confirm that by accident we didn't outrule a valid notation.